server/src/main/java/io/crate/expression/symbol/Aggregation.java

```diff
@@ -151,22 +150,20 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-        Aggregation that = (Aggregation) o;
-        return Objects.equals(signature, that.signature) &&
-               Objects.equals(inputs, that.inputs) &&
-               Objects.equals(valueType, that.valueType) &&
-               Objects.equals(filter, that.filter);
+        return o instanceof Aggregation that &&
+            Objects.equals(signature, that.signature) &&
+            Objects.equals(inputs, that.inputs) &&
+            Objects.equals(valueType, that.valueType) &&
+            Objects.equals(filter, that.filter);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(signature, inputs, valueType, filter);
+        int result = signature.hashCode();
```

Comment:
Maybe add a comment here to say that this is explicitly not calling Objects.hash to avoid the vararg array allocation?  IDE's will often recommend changing these sort of hashCode implementations to use Objects.hash() so a comment should prevent anyone from switching them back again.

Reply:
> IDE's will often recommend changing these sort of hashCode implementations to use Objects.hash() so a comment should prevent anyone from switching them back again.

Which IDEs warn about this? If it's intellij we could disable the inspection.

Not sure we want to start adding comments on all hashCode implementations that don't use `Objects.hash`. 
We could also consider adding it to forbidden signature, but there are many more places using it.

Reply:
Looking again, I'm wrong about this, at least for IntelliJ - there are a few `31 * xxx` hashcode implementations which aren't showing warnings for me.  So never mind, go ahead and merge as is :)